### PR TITLE
Expand documentation

### PR DESCRIPTION
**Pull request by ianprime0509**
Addresses #576 and (partially) #454.

First of all, thank you for all your work on this project!  I recently started a project using react-modal, and I picked up a lot of information from issue comments and such that I thought would be good to include in the main documentation as well.  Please let me know if there's anything you'd like me to change!

Also, I noticed in #454 that moving documentation from the README to the book was under consideration.  If you'd like, I can update this pull request or make another one to do that, since it looks like there are only a few things that the README discusses in more detail.

Changes proposed:
* Add more information about styling CSS classes and transitions
* Add section on accessibility, including app element details (addresses #576)
* Create missing `examples/README.md` and `contributing/README.md`
* Reorganize initial book section (with subsection labels)

Upgrade Path (for changed or removed APIs): N/A

Acceptance Checklist:
- [X] All commits have been squashed to one.
- [X] The commit message follows the guidelines in `CONTRIBUTING.md`.
- [X] Documentation (README.md) and examples have been updated as needed.
- [X] If this is a code change, a spec testing the functionality has been added.
- [X] If the commit message has [changed] or [removed], there is an upgrade path above.

### CODE REVIEW

File: docs/accessibility/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+title as well as a longer description:
+
+```jsx
+<Modal
+  isOpen={modalIsOpen}
+  aria={{
+    labelledby: "heading",
+    describedby: "full_description"
+  }}>
+  <h1 id="heading">H1</h1>
```
Comment (diasbruno) [inline docs/accessibility/README.md]: `H1` => `Information` or `Alert`.

Just to look more realistic.

**Comment by diasbruno** [comment]
First, thank you so much for improving the docs. 

Regarding the `README.md`, we can integrate in a second PR, just to make things simple.

**Comment by ianprime0509** [comment]
No problem!  I fixed the issue you mentioned and re-squashed everything.  Just to clarify, for the README, should I make another PR based off this one and submit it separately, or something else?

**Comment by diasbruno** [comment]
Another PR would be better. Give a thumbs up if this one is ready to be merged.

**Pull request merged**